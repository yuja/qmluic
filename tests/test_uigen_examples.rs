pub mod common;

#[test]
fn test_gadget_properties() {
    insta::assert_snapshot!(common::translate_file("examples/GadgetProperties.qml").unwrap());
}

#[test]
fn test_item_views() {
    insta::assert_snapshot!(common::translate_file("examples/ItemViews.qml").unwrap());
}

#[test]
fn test_layout_flow() {
    insta::assert_snapshot!(common::translate_file("examples/LayoutFlow.qml").unwrap());
}

#[test]
fn test_main_window() {
    insta::assert_snapshot!(common::translate_file("examples/MainWindow.qml").unwrap());
}

#[test]
fn test_settings_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/SettingsDialog.qml").unwrap());
}

#[test]
fn test_static_item_model() {
    insta::assert_snapshot!(common::translate_file("examples/StaticItemModel.qml").unwrap());
}

#[test]
fn test_various_layouts() {
    insta::assert_snapshot!(common::translate_file("examples/VariousLayouts.qml").unwrap());
}

#[test]
fn test_customwidget_common_my_dialog_button_box() {
    insta::assert_snapshot!(common::translate_file(
        "examples/customwidget/common/MyDialogButtonBox.qml"
    )
    .unwrap());
}

#[test]
fn test_customwidget_main_dialog() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/MainDialog.qml").unwrap()
    );
}

#[test]
fn test_customwidget_settings_form() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/SettingsForm.qml").unwrap()
    );
}

#[test]
fn test_thg_hg_email_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/thg/HgEmailDialog.qml").unwrap());
}
