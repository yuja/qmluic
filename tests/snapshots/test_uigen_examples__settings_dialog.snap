---
source: tests/test_uigen_examples.rs
expression: "common::translate_file(\"examples/SettingsDialog.qml\").unwrap()"
---
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QDialog" name="dialog">
  <property name="geometry">
   <rect>
    <height>300</height>
    <width>400</width>
    <x>0</x>
    <y>0</y>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item column="0" row="0">
      <widget class="QLabel" name="hgExecutableLabel">
       <property name="text">
        <string>Hg executable:</string>
       </property>
      </widget>
     </item>
     <item column="1" row="0">
      <layout class="QHBoxLayout" name="hgExecutableLayout">
       <item>
        <widget class="QLineEdit" name="hgExecutableEdit">
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="hgExecutableBrowseButton">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item column="0" row="1">
      <widget class="QLabel" name="textFontLabel">
       <property name="text">
        <string>Text font:</string>
       </property>
      </widget>
     </item>
     <item column="1" row="1">
      <layout class="QHBoxLayout" name="textFontLayout">
       <item>
        <widget class="QFontComboBox" name="textFontFamilyEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="textFontSizeEdit">
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="textFontSizeLabel">
         <property name="text">
          <string>pt</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation" stdset="0">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <height>40</height>
       <width>20</width>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
</ui>

